becquerel/core/utils.py

```diff
@@ -100,7 +101,10 @@ def handle_datetime(input_time, error_name="datetime arg", allow_none=False):
         )
         return datetime.datetime(input_time.year, input_time.month, input_time.day)
     elif isinstance(input_time, str):
-        return dateutil_parse(input_time)
+        try:
+            return dateutil_parse(input_time)
+        except ParserError:
+            return datetime.datetime.strptime(input_time, "%Y_%m_%d_%H_%M_%S")
```

Comment:
If this fails, does it raise a useful error? If not we should also wrap this in a `try... except`.

Reply:
Yes, if `strptime` fails due to an incorrect format that `dateutil_parse` also couldn't handle first, we get
```py
ValueError: time data '2021_01_01-12_01_01' does not match format '%Y_%m_%d_%H_%M_%S'
```

Reply:
As discussed, please update the docstring, including the `Raises` part to reflect the current behavior.